src/cache/cache_las.c

```diff
@@ -572,8 +608,23 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
          */
         WT_ASSERT(session, modifies.size > 0);
         __wt_modify_vector_pop(&modifies, &upd);
-        /* The key didn't exist back then, which is globally visible. */
-        WT_ASSERT(session, upd->type == WT_UPDATE_STANDARD || upd->type == WT_UPDATE_TOMBSTONE);
+
+        /* If we popped a modify then it should be flagged as in the history store. */
+        if (upd->type == WT_UPDATE_MODIFY) {
+            WT_ASSERT(session, F_ISSET(upd, WT_UPDATE_HISTORY_STORE));
+            __wt_modify_vector_peek(&modifies, &prev_upd);
+            /*
+             * Retrieve the full value of the modify from the history store. This avoid us having to
+             * iterate the full update list associated with the modify and recalculating the reverse
+             * deltas.
+             */
+            WT_ERR(__las_get_value(
```

Comment:
I just had a thought since this made me wonder. If this modify is on the same lookaside page that we're writing updates to then it makes sense, but if we had to read a page into cache to do this, isn't it likely that this is actually worse?

Reply:
It's the same key so likely we are on the same page here.

Reply:
Yeah, I realize it's probably going to be the same page but it's not guaranteed. And in the case where it isn't, it's going to be very costly no?

Hard to say how often each scenario occurs vs the cost so no need to address it unless there are issues later.